src/policy_provider/SamplePolicyProvider.c

```diff
@@ -408,11 +420,14 @@ int BSLP_PolicyRule_Init(BSLP_PolicyRule_t *self, const char *desc, BSLP_PolicyP
 {
     ASSERT_ARG_NONNULL(self);
     memset(self, 0, sizeof(*self));
-    string_init_set_str(self->description, desc);
-    self->sec_block_type    = sec_block_type;
-    self->target_block_type = target_block_type;
-    self->predicate         = predicate;
-    self->context_id        = context_id;
+    size_t desc_sz    = strnlen(desc, POLICY_RULE_DESCRIPTION_MAX_STRLEN);
```

Comment:
It could be helpful to visibly separate this text handling block with newlines before/after.